### ISSUE TITLE
Line rebreaking on terminal resize

### ISSUE THREAD

**Issue by GoogleCodeExporter** [mintty/mintty#82]
```
It would be rather nifty if long lines could be rebroken after the number
of terminal columns changes, much like you see with flowing text in a web
browser. Apple's Terminal.app does this.

This would also solve the problem of window contents disappearing when
increasing the font size in a window that can't be enlarged.

(Fullscreen apps such as editors usually already do this via the SIGWINCH
signal, so this primarily concerns plain command line output and the
scrollback.)
```

Original issue reported on code.google.com by `andy.koppe` on 4 Apr 2009 at 7:24

**Comment by stealthybox** [comment]
It doesn't solve the issue, but it may be helpful for somebody:

I use tmux with gitbash and zsh.
tmux will dynamically rewrap lines for you.
(in addition to splitting panes and all kinds of other fun stuff)

It takes some configuration to get working nicely, but it's definitely worth it.

You can use pacman with with the [git-for-windows-sdk](https://github.com/git-for-windows/build-extra/releases) to install tmux: (`pacman -S tmux`)

**Comment by silverwind** [comment]
Got to add my vote here, this would be a very useful feature.

Even `cmd` has it nowadays via the "wrap text output on resize" option:

![win10wrap](https://user-images.githubusercontent.com/115237/44956732-fad8bb00-aec8-11e8-86e1-06affd45013f.gif)

**Comment by mintty** [comment]
Your vote is noted, but it is quite unlikely that this feature will be added to mintty.
Very few terminals provide such a feature and there are a bunch of design issues to be clarified in the first place. Also I estimate it would be a *lot* of effort.

**Comment by Biswa96** [comment]
@silverwind It's magic of ConHost.exe console executable and ConDrv.sys driver. CMD is just a shell.

**Comment by mintty** [comment]
@silverwind, what's actually the benefit of such ragged wrap-around into a narrow terminal? I don't think the resulting display is very readable and thus not useful either, like `ls /` and then squeezing the terminal (tested with vte):
```
A	    cygdrive		 dev   M
	 proc		T    W
bin	    Cygwin.bat		 etc   makefil
e  S		tmp  Y
C	    Cygwin.ico		 home  mnt
	 sbin		usr
Cygwin-Terminal.ico  lib   O	 setup-x86.exe	var
```

**Comment by silverwind** [comment]
I often have multiple narrow windows up and if I run a command that results in bad wrapping, I have to resize and then run the command again to get proper wrapping, which is rather annoying, especially if the command in question is long-running.

**Comment by mintty** [comment]
That's not quite matching my question. It was about resizing the window and then having bad wrapping *after* the resize, maybe even as a consequence of the resizing. I guess this is the more frequent situation.

**Comment by silverwind** [comment]
Also kind of related here is one issue with copy after terminal resize. Steps to reproduce:

1. Have a small terminal
2. Echo a string that will wrap, like

````
loooooooooooooooooooooooooooooooooooooooooong
````
3. Resize the terminal to a larger width
4. Select and copy the string
5. The copied string now contains extraneous whitespace:
````
loooooooooooooooooooooooooooooooooooo         oooooong
````

**Comment by silverwind** [comment]
Thanks, can confirm this working!

**Comment by mintty** [comment]
Released 2.9.1.

**Comment by mintty** [comment]
Released 2.9.2. Fixed the fix for the looong issue.

**Comment by silverwind** [comment]
Thanks, much appreciated.

**Comment by Abdull** [comment]
I'm using mintty 2.9.6, but resizing the mintty window does not re-layout previously written text.

**Comment by mintty** [comment]
If it did, this issue would be close, so what's the point?

**Comment by mintty** [comment]
Released 3.6.0.

**Comment by mintty** [comment]
Released 3.6.1.
